service/history/replication/executable_activity_state_task.go

```diff
@@ -49,11 +50,17 @@ type (
 		definition.WorkflowKey
 		ExecutableTask
 		req *historyservice.SyncActivityRequest
+
+		// following fields are used only for batching functionality
+		batchLock     sync.Mutex
+		batchable     bool
```

Comment:
Mind separating this field from the mutex? I think we should only group mutexes with the fields they protect